### PR TITLE
Node Execute

### PR DESCRIPTION
**Pull request by elliott-davis**
ping @chef-cookbooks/delivery-migration-team

### CODE REVIEW

**Comment by chef-delivery** [comment]
This PR has passed 'Verify' and is ready for review and approval!
  Use: '@delivery approve' when code review is complete.

File: libraries/node_install.rb [hunk 1]
```diff
@@ -18,32 +18,39 @@
...
+require 'mixlib/shellout'
 
 class Chef
   class Resource::NodeInstall < ChefCompat::Resource
     resource_name :node_install
 
-    property :node_version, kind_of: String, name_property: true
-    property :prefix, kind_of: String, default: '/usr/local/bin'
+    property :version, kind_of: String, name_property: true
+    prefix = '/opt/languages/node'
```
Comment (schisamo) [inline libraries/node_install.rb]: Don't we want the following?

```ruby
property :prefix, kind_of: String, default: '/opt/languages/node'
```

File: libraries/node_execute.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    property :command, kind_of: String, name_property: true
+    property :version, kind_of: String
+
+    # Useful propertys from the `execute` resource that might need overriding
+    property :cwd, kind_of: String
+    property :environment, kind_of: Hash, default: {}
+    property :user, kind_of: [String, Integer]
+    property :sensitive, kind_of: [TrueClass, FalseClass], default: false
+
+    prefix = '/opt/languages/node'
```
Comment (schisamo) [inline libraries/node_execute.rb]: Don't we want the following?

```ruby
property :prefix, kind_of: String, default: '/opt/languages/node'
```

Comment (elliott-davis) [inline libraries/node_execute.rb]: I would say no - I don't think users should have control over where we put node on the system - this also increases the chance that we end up with multiple nodes of the same version all over the box.  Node is just the engine - nobody can affect it's state.

**Comment by chef-delivery** [comment]
This PR has passed 'Verify' and is ready for review and approval!
  Use: '@delivery approve' when code review is complete.

**Comment by schisamo** [comment]
@delivery approve

**Comment by chef-delivery** [comment]
Change: f8e7df96-ba42-40ee-9432-949f7c0db56f approved by: @schisamo

**Comment by chef-delivery** [comment]
This PR has passed 'Acceptance' and is ready to be delivered!
  Use: '@delivery deliver' when validated in acceptance.

**Pull request merged**